.xo-config.json

```diff
@@ -16,6 +16,8 @@
 		"no-void": "off",
 		"react/jsx-key": "off",
 		"unicorn/no-array-callback-reference": "off",
+		"@typescript-eslint/no-non-null-assertion": "off",
+		"@typescript-eslint/no-non-null-asserted-optional-chain": "off",
```

Comment:
What's the problem again? It worked until now and https://github.com/typescript-eslint/typescript-eslint/issues/1976 was resolved

Reply:
Same issue

Reply:
If Sindre re-enabled them they're probably working now, so maybe our code is wrong.

The only reason I can think of to disable these are if the changes required just clutter the code without actually improving it

Reply:
`@typescript-eslint/no-non-null-assertion` is correct in being disabled, we need those errors.

I'm checking the other one

Reply:
`@typescript-eslint/no-non-null-asserted-optional-chain` should probably not be disabled.

However I'm not liking most of the type changes in this PR so I'd rather avoid all of them until a TS developer can review them properly. 

If you're going to update XO:

- disable `no-unsafe-argument` + a TODO to review later
- disable `no-non-null-asserted-optional-chain` + same TODO
- disable `no-non-null-assertion` with a comment that we need those runtime errors

Reply:
Ok

Can I leave the `interface` ones?

Reply:
We are using a JSON file which makes it not possible to leave a comment.

Reply:
Indeed XO does not support JSON with comments. Can you convert the config to .yml?

Reply:
Do you have any sample repo that I can follow? I have never seen a yml xo file.

Reply:
@fregante in case you missed this comment

Reply:
Find a JSON to YML converter

Reply:
That did not work.

I am going to revert the Type's and just update dep

Reply:
You're right, I assumed it worked since ESLint supports it: https://github.com/xojs/xo/issues/536

Reply:
> We are using a JSON file which makes it not possible to leave a comment.

That's not entirely true: https://github.com/sindresorhus/sindre-playground/blob/565ba9c61dbf4b13d587bcae7047120952f924c5/package.json#L26-L31